### PR TITLE
Add the backward support for QR

### PR DESCRIPTION
**Pull request by aoyulong**
### PR types
Others

### PR changes
OPs

### Describe

The PR adds the backward support for QR decomposition.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_qr_op.py [hunk 1]
```diff
@@ -21,6 +21,107 @@
...
+
+    def get_input_and_output(self):
+        dtype = self.get_dtype()
+        shape = self.get_shape()
+        mode = self.get_mode()
+        assert mode != "r", "Cannot be backward in r mode."
+        a = np.random.rand(*shape).astype(dtype)
+        # a = np.array([[1, 2, 3, 2], [4, 5, 6, 3], [7, 8, 7, 4]]).astype(dtype)
+        # a = np.array([[1, 2], [3, 4], [5, 6]]).astype(dtype)
+        # a = np.array([[[1, 2], [3, 4], [5, 6]], [[1, 2], [3, 4], [5, 6]]]).astype(dtype)
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/test_qr_op.py]: no need this annotation?

File: python/paddle/fluid/tests/unittests/test_qr_op.py [hunk 2]
```diff
@@ -21,6 +21,107 @@
...
+        self.check_output()
+
+    def test_check_grad_normal(self):
+        # dQ = np.array([[1, 1, 1], [1, 1, 1], [1, 1, 1]]).astype(self.dtype)
+        # dR = np.array([[1, 1, 1, 1], [1, 1, 1, 1], [1, 1, 1, 1]]).astype(self.dtype)
+        # dQ = np.array([[1, 1], [1, 1], [1, 1]]).astype(self.dtype)
+        # dR = np.array([[1, 1], [1, 1]]).astype(self.dtype)
+        # dQ = np.array([[[1, 1], [1, 1], [1, 1]], [[1, 1], [1, 1], [1, 1]]]).astype(self.dtype)
+        # dR = np.array([[[1, 1], [1, 1]], [[1, 1], [1, 1]]]).astype(self.dtype)
+        # self.check_grad(['X'], ['Q', 'R'], user_defined_grad_outputs=[dQ, dR])
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/test_qr_op.py]: no need this annotation?

**Review by XieYunshen — DISMISSED** [review]
LGTM for `set_tests_properties(test_qr_op PROPERTIES TIMEOUT 60)`

File: python/paddle/fluid/tests/unittests/test_qr_op.py (hunk 1 above)
Comment (aoyulong) [reply python/paddle/fluid/tests/unittests/test_qr_op.py]: Done.

File: python/paddle/fluid/tests/unittests/test_qr_op.py (hunk 2 above)
Comment (aoyulong) [reply python/paddle/fluid/tests/unittests/test_qr_op.py]: Done.

**Review by XieYunshen — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]
LGTM

**Pull request merged**